RELEASE/scripts/autoscend/auto_consume.ash

```diff
@@ -1223,7 +1223,20 @@ ConsumeAction auto_bestNightcap()
 	int best = 0;
 	for(int i=1; i < count(actions); i++)
 	{
-		if(desirability(i) > desirability(best)) best = i;
+		if(desirability(i) < desirability(best))
+		{
+			// This consumable is less desirable than the best consumable found so far
+			continue;
+		}
+
+		if(desirability(i) == desirability(best) && mall_price(actions[i].it) >= mall_price(actions[best].it))
```

Comment:
use `historical_price` not `mall_price` also this checks everything regardless of acquisition method not just pulls which will cause issues.

Reply:
I asked which to use in the discord and Rinn said `mall_price` was fine. Should I change it?

Can you elaborate on what issues this could cause? What do you suggest changing?

Reply:
mall prices might prompt a server hit for each item. to prevent it we already do once a day
```
		auto_log_info("Bulk caching mall prices for consumables");
		mall_prices("food");
		mall_prices("booze");
		mall_prices("hprestore");
		mall_prices("mprestore");
```
as part of our initialization. which gets all the food and drink prices with the minimum amount of server hits.

If our defensive measure quoted above works then there is no difference between historical price and mall price, since to prevent abuse mall price automatically uses historical price if you already got the price of an item today.

But if for some reason it does not work as intended then you will generate countless server hits that will also take a very long time. For such a bulk operation it is probably a good idea to just use historical price in the first place and rely on the initialization update of historical prices to keep those up to date.

Reply:
Something occured to me. We only update the mall prices when not in hardcore.
someone who always does hardcore runs with autoscend could have really outdated figures.
... that said, its not that big a deal. I think historical price is still best here

Reply:
Yeah, even if the prices are way off it's still less arbitrary than taking the first in the list when tied

Reply:
> Can you elaborate on what issues this could cause? What do you suggest changing?

should be checking if it `i` and `best` are both pulled. Otherwise you could get into a situation where we have something in inventory (or we will buy from an NPC) which is equal to something we're going to pull but we decide to pull just because the mall price is lower.

Reply:
The previous behavior did not account for that, either. It just took whichever came first.
Choosing an item on-hand would be preferred, but adding that logic seems out of scope for this PR.